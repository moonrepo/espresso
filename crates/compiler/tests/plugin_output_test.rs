mod utils;

use espresso_common::EsTarget;
use espresso_package::Package;
use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::*;

mod plugin_output {
    use super::*;

    #[tokio::test]
    async fn adds_mjs_ext_to_imports_exports() {
        let sandbox = create_sandbox("imports-exports");
        let package = Package::new(sandbox.path()).unwrap();
        let compiler = create_compiler(sandbox.path(), &package);

        let out_dir = compiler.compile(EsTarget::Es2020).await.unwrap();

        assert_snapshot!(read_file(out_dir.join("index.mjs")));
    }
}
