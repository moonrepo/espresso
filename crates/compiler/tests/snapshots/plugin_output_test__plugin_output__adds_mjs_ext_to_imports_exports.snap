---
source: crates/compiler/tests/plugin_output_test.rs
expression: "read_file(out_dir.join(\"index.mjs\"))"
---
// Imports
import 'module';
import "./file.mjs";
import './file.mjs';
// Dynamic imports
await import('module');
await import("./nested/file.mjs");
await import("../parent/file.mjs");
// Exports
export * from "./file.mjs";
export { foo, bar } from "./file.mjs";
export * as ns from "./file/../weird/path.mjs";

